app/js/views/themes.jsx

```diff
@@ -51,24 +30,12 @@ export default class Themes extends React.Component {
   }
 
   render() {
-    const themeItems = this.state.themes.map((theme, id) => {
-      let itemClassName = 'themes-list__item';
-      if (!theme.enabled) {
-        itemClassName += ' themes-list__item--deactivated';
-      }
-
-      return (
-        <li key={id}
-            className={itemClassName}>
-          <input className="themes-list__toggle" type="checkbox"
-                 checked={theme.enabled}
-                 onChange={this.handleOnChange.bind(this, id)}/>
-          <span className="themes-list__name">{theme.name}</span>
-          <button className="themes-list__remove"
-                  onClick={this.handleOnDelete.bind(this, id)}></button>
-        </li>
-      );
-    });
+    const themeItems = this.state.themes.map(theme => (
+      <ThemesListItem key={theme.id}
+                      theme={theme}
+                      update={this.update}
```

Comment:
Surprisingly passing a cb as a prop is a common pattern in React, see https://facebook.github.io/react/tips/communicate-between-components.html
They recommend an event system when the components don't share a parent-child relationship.

I'm leaving it for now, and we'll change the code when we adopt an event approach.